### ISSUE TITLE
Ability to customize ports to scrape for kubelet (a.k.a "Prometheus failed to scrape 50% of kubelets.")

### ISSUE THREAD

**Issue by desaintmartin** [coreos/prometheus-operator#1933]
Hello,

Today, it seems prometheus-operator generates several headless services / endpoints, kubelet (in the kube-system namespace) being one of them.

Prometheus operator does not seem to care if http-metrics port (a.k.a insecure port, a.k.a read-only port) is enabled or disabled and generates an endpoint containing 3 ports: http-metrics (10255), cadvisor (4194) and https-metrics (10250).

See https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/operator.go#L530

Is there a way to customize it?

My kube-prometheus tries to scrape this non existent http-metrics port, causing alertmanager to send messages like "Prometheus failed to scrape 50% of kubelets.". Removing the port from the endpoint make it disappear from the Prometheus targets, but only until prometheus-operator creates it back.

**Comment by brancz** [comment]
This should just be a matter of choosing the right port in the `ServiceMonitor` object, and by default all objects in this repo use the secure port. Can you share the `ServiceMonitor` you use to monitor the kubelets?

**Comment by desaintmartin** [comment]
Closing as false positive.

For the record, I had another custom `ServiceMonitor` also scrapping `kube-system` namespace using http-metrics port with malformed `selector`.

It made ALL services matching 10255 port be scrapped. Since the code linked in the original description defines such a port, Prometheus-operator tried to scrape it as well.

I got initially confused because of this hardcoded kubelet `Service` creation.
